x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicy.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.enrich;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents an enrich policy including its configuration.
+ */
+public final class EnrichPolicy implements Writeable, ToXContentObject {
+
+    static final ParseField TYPE = new ParseField("type");
+    static final ParseField QUERY = new ParseField("query");
+    static final ParseField INDEX_PATTERN = new ParseField("index_pattern");
+    static final ParseField ENRICH_KEY = new ParseField("enrich_key");
+    static final ParseField ENRICH_VALUES = new ParseField("enrich_values");
+    static final ParseField SCHEDULE = new ParseField("schedule");
+
+    @SuppressWarnings("unchecked")
+    static final ConstructingObjectParser<EnrichPolicy, Void> PARSER = new ConstructingObjectParser<>("policy",
+        args -> {
+            return new EnrichPolicy(
+                Type.read((String) args[0]),
+                (QuerySource) args[1],
+                (String) args[2],
+                (String) args[3],
+                (List<String>) args[4],
+                (TimeValue) args[5]
+            );
+        }
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), TYPE);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> {
```

Comment:
How do we feel about making the `query` field an optional constructor arg, with the assumption that an empty field is just a `match_all`? My thought is that it aligns with the `_search` action since an empty query just returns all documents